### PR TITLE
removed almost all URLParse usage

### PR DESCRIPTION
**Pull request by pmcb55**
# Internal refactor (and breaking interface changes!)

Still 4 cases of using URLParse, but those are all internal and can be replaced with the browser URL class in a separate PR (this PR should clean up all the interfaces at least).

This ultimately comes from this issue: https://github.com/inrupt/solid-client-authn-js/issues/420

### CODE REVIEW

File: packages/browser/src/util/Fetcher.ts [hunk 1]
```diff
@@ -27,22 +27,23 @@
 /**
  * A wrapper method to wrap the standard w3 fetch library
  */
-import URL from "url-parse";
 import _fetch from "cross-fetch";
 
 export interface IFetcher {
-  fetch(url: RequestInfo | URL, init?: RequestInit): Promise<Response>;
+  fetch(url: RequestInfo | string, init?: RequestInit): Promise<Response>;
```
Comment (NSeydoux) [inline packages/browser/src/util/Fetcher.ts:33]:
```suggestion
  fetch(url: RequestInfo, init?: RequestInit): Promise<Response>;
```
`RequestInfo` is already a `Request | string`

File: packages/browser/src/util/Fetcher.ts [hunk 2]
```diff
@@ -27,22 +27,23 @@
...
 }
 
 /**
  * @hidden
  */
 export default class Fetcher implements IFetcher {
-  async fetch(url: RequestInfo | URL, init?: RequestInit): Promise<Response> {
-    const fetchUrl = url instanceof URL ? url.toString() : url;
+  async fetch(
+    url: RequestInfo | string,
```
Comment (NSeydoux) [inline packages/browser/src/util/Fetcher.ts:41]:
```suggestion
    url: RequestInfo,
```

**Review by NSeydoux — APPROVED** [review]

File: packages/browser/src/ClientAuthentication.ts [hunk 3]
```diff
@@ -55,14 +54,9 @@ export default class ClientAuthentication {
...
-      if (typeof url !== "string") {
-        return url;
-      }
-      return new UrlParse(url);
-    }
-    return undefined;
+  private urlOptionToUrl(url?: string): string | undefined {
+    // TODO: PMcB55: Validate URL.
+    return url;
   }
```
[the remark this answers is not in the document]
Comment (pmcb55) [reply packages/browser/src/ClientAuthentication.ts:60]: Well, my intent was to just rename this function (to just be an URL validating function), possibly move it to a utils file if is was useful elsewhere. I also want to look into adding proper error handling for malformed URLs, as I'm a bit concerned the code never handled that.

File: packages/browser/src/login/oidc/IssuerConfigFetcher.ts [hunk 4]
```diff
@@ -138,30 +137,34 @@ export default class IssuerConfigFetcher implements IIssuerConfigFetcher {
...
     return (parsedConfig as unknown) as IIssuerConfig;
   }
 
-  async fetchConfig(issuer: URL): Promise<IIssuerConfig> {
+  async fetchConfig(issuer: string): Promise<IIssuerConfig> {
     let issuerConfig: IIssuerConfig;
 
-    const wellKnownUrl = new URL(issuer.toString());
-    wellKnownUrl.set("pathname", "/.well-known/openid-configuration");
+    const wellKnownUrl = `${issuer}${
```
[the remark this answers is not in the document]
Comment (pmcb55) [reply packages/browser/src/login/oidc/IssuerConfigFetcher.ts:164]: Yep, absolutely - that was my intent, but I should have added the marker comment - cheers.

File: packages/browser/src/login/oidc/OidcLoginHandler.ts [hunk 5]
```diff
@@ -57,28 +68,25 @@ export default class OidcLoginHandler implements ILoginHandler {
...
   }
 
   async handle(options: ILoginOptions): Promise<void> {
-    // Check to ensure the login options are correct
-    const optionsError: Error | null = this.checkOptions(options);
-    if (optionsError) {
-      throw optionsError;
+    if (!hasIssuer(options)) {
+      throw new ConfigurationError(
+        "OidcLoginHandler requires an OIDC issuer: 'oidcIssuer'"
```
Comment (NSeydoux) [inline packages/browser/src/login/oidc/OidcLoginHandler.ts:78]:
```suggestion
        `OidcLoginHandler requires an OIDC issuer: missing property 'oidcIssuer' in ${JSON.stringify(options)}`
```

File: packages/browser/src/login/oidc/OidcLoginHandler.ts [hunk 6]
```diff
@@ -57,28 +68,25 @@ export default class OidcLoginHandler implements ILoginHandler {
...
+      );
+    }
+    if (!hasRedirectUrl(options)) {
+      throw new ConfigurationError(
+        "OidcLoginHandler requires a redirect URL: 'redirectUrl'"
```
Comment (NSeydoux) [inline packages/browser/src/login/oidc/OidcLoginHandler.ts:83]:
```suggestion
        `OidcLoginHandler requires a redirect URL: missing property 'redirectUrl' in ${JSON.stringify(options)}`
```

**Review by NSeydoux — APPROVED** [review]

**Pull request merged**